src/libraries/System.Reflection.Metadata/tests/Metadata/Decoding/CustomAttributeDecoderTests.cs

```diff
@@ -87,6 +89,258 @@ public void TestCustomAttributeDecoder()
+                            Assert.Equal("int32", value.FixedArguments[1].Type);
+                            Assert.Equal(12, value.FixedArguments[1].Value);
+
+                            Assert.Equal(1, value.NamedArguments.Length);
+
+                            Assert.Equal("string", value.NamedArguments[0].Type);
+                            Assert.Equal("Bye", value.NamedArguments[0].Value);
+                            break;
+
+                        case 4:
+                            Assert.Equal(1, value.FixedArguments.Length);
+
+                            Assert.Equal("uint8", value.FixedArguments[0].Type);
+                            Assert.Equal((byte)1, value.FixedArguments[0].Value);
+
+                            Assert.Equal(1, value.NamedArguments.Length);
+
+                            Assert.Equal("uint8", value.NamedArguments[0].Type);
+                            Assert.Equal((byte)2, value.NamedArguments[0].Value);
+                            break;
+
+                        case 5:
+                            Assert.Equal(2, value.FixedArguments.Length);
+
+                            Assert.Equal("bool", value.FixedArguments[0].Type);
+                            Assert.Equal(true, value.FixedArguments[0].Value);
+                            Assert.Equal("int32", value.FixedArguments[1].Type);
+                            Assert.Equal(13, value.FixedArguments[1].Value);
+
+                            Assert.Empty(value.NamedArguments);
+                            break;
+
+                        case 6:
+                            Assert.Equal(1, value.FixedArguments.Length);
+
+                            Assert.Equal("System.Reflection.Metadata.Decoding.Tests.CustomAttributeDecoderTests/MyEnum", value.FixedArguments[0].Type);
+                            Assert.Equal(MyEnum.Property, Enum.ToObject(typeof(MyEnum), value.FixedArguments[0].Value));
+
+                            Assert.Empty(value.NamedArguments);
+                            break;
+
+                        case 7:
+                            Assert.Equal(1, value.FixedArguments.Length);
+
+                            Assert.Equal("[System.Runtime]System.Type", value.FixedArguments[0].Type);
+                            Assert.Equal(typeof(HasAttributes).FullName, value.FixedArguments[0].Value);
+
+                            Assert.Empty(value.NamedArguments);
+                            break;
+
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.HasAssemblyFiles))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/60579", TestPlatforms.iOS | TestPlatforms.tvOS)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Type assembly name is different on .NET Framework.")]
+        public void TestCustomAttributeDecoderGenericUsingReflection()
```

Comment:
This test testing exact same scenarios as above, but using reflection instead of hard coding. Added this test for comparison because there is a comment and an open issue for adding more tests and refactoring existing one:
https://github.com/dotnet/runtime/blob/bc64ffee79272691471a07aa6855fcb892096bf2/src/libraries/System.Reflection.Metadata/tests/Metadata/Decoding/CustomAttributeDecoderTests.cs#L80-L83 
I can add more tests and refactor that existing test to use reflection if needed, but personally I prefer hard coded version as it more clear for troubleshooting. Please let me know which test is preferred

Reply:
Update: left one of the hard coded test as is, for other tests used generic approach